modules/launcher/src/main/java/org/mule/runtime/module/launcher/service/LifecycleFilterServiceProxy.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+
+package org.mule.runtime.module.launcher.service;
+
+import static org.mule.runtime.core.util.Preconditions.checkArgument;
+import static org.mule.runtime.module.launcher.util.ReflectionUtils.findImplementedInterfaces;
+import org.mule.runtime.api.service.Service;
+import org.mule.runtime.core.api.lifecycle.Startable;
+import org.mule.runtime.core.api.lifecycle.Stoppable;
+
+import java.lang.reflect.InvocationHandler;
+import java.lang.reflect.Method;
+import java.lang.reflect.Proxy;
+
+/**
+ * Proxies a {@link Service} instance to filter invocations of lifecycle methods from
+ * {@link Startable} and {@link Stoppable} interfaces.
+ */
+public class LifecycleFilterServiceProxy implements InvocationHandler
```

Comment:
The proxy is required not because of the lifecycle applied by mule/spring, but because a user can get the service instance from the registry and manually invoke those methods